activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/service/ModelServiceImpl.java

```diff
@@ -401,19 +399,28 @@ public Model importModel(Project project, ModelType modelType, FileContent fileC
             )
         );
 
-        Model model = importModelFromContent(project, modelType, fileContent);
-        return model;
+        return importModelFromContent(project, modelType, fileContent);
     }
 
     @Override
-    public Model importModelFromContent(Project project, ModelType modelType, FileContent fileContent) {
+    public ImportedModel importModelFromContent(Project project, ModelType modelType, FileContent fileContent) {
         throwExceptionIfFileIsExecutable(modelType.getName(), fileContent);
-        Model model = null;
-        if (modelTypeService.isJson(modelType) || isJsonContentType(fileContent.getContentType())) {
-            model = convertContentToModel(modelType, fileContent);
-        } else {
-            model = createModelFromContent(modelType, fileContent);
+        Model model = loadModel(modelType, fileContent);
+        String originalId = resolveOriginalId(modelType, fileContent, model);
+
+        model.setScope(ModelScope.PROJECT);
+        createModel(project, model);
+        if (originalId != null) {
+            return new ImportedModel(model, originalId, buildModelTypeAwareIdentifier(model));
         }
+        return new ImportedModel(model);
+    }
+
+    private static String buildModelTypeAwareIdentifier(Model model) {
```

Comment:
nitpicking: does it need to be static since it's only used by non-static class methods?

Reply:
good catch: I didn't pay attention that I had the option `Make static` checked by default in my IDE for methods extractions. I'm going to remove the static modifier.